topology/probes/opencontrail.go

```diff
@@ -100,27 +100,30 @@ func (mapper *OpenContrailProbe) retrieveMetadata(metadata graph.Metadata, itf c
 // for instance, the tap is first created by nova and this information
 // is then propagated to contrail. We then retry to get interface from
 // contrail introspect with a delay between each attempt.
-func getInterfaceFromIntrospect(host string, port int, name string) (collection.Collection, collection.Element, error) {
-	var err error
+func getInterfaceFromIntrospect(host string, port int, name string) (col collection.Collection, elem collection.Element, err error) {
 	try := 3
 	delay := 500 * time.Millisecond
 
 	for i := 0; i < try; i++ {
-		col, e := collection.LoadCollection(descriptions.Interface(), []string{fmt.Sprintf("%s:%d", host, port)})
-		err = e
-		if e == nil {
-			itf, e := col.SearchStrictUnique(name)
-			err = e
-			if e == nil {
-				return col, itf, e
-			}
+		if i > 0 {
```

Comment:
nit: you could use common.Retry

Reply:
common.Retry expect a function that returns only an error, I don't see how I can use it here

Reply:
That is not important ;) It seems to me you are implementing the exact same pattern. That could be something like this

```
fnc := func() error {
  col, err = collection.LoadCollection(descriptions.Interface(), []string{fmt.Sprintf("%s:%d", host, port)})
  if err != nil {
    return err
  }

  elem, err = col.SearchStrictUnique(name)
  if err != nil {
    col.Close()
    return err
  }

  return nil
}

common.Retry(fnc, 3, 500*timeMillisecond)
```